connect/runtime/src/main/java/org/apache/kafka/connect/runtime/WorkerSinkTask.java

```diff
@@ -237,6 +239,19 @@ public String toString() {
                 '}';
     }
 
+    private ConsumerRecords<byte[], byte[]> pollConsumer(long timeoutMs) {
+        ConsumerRecords<byte[], byte[]> msgs = consumer.poll(timeoutMs);
+
+        // Exceptions raised from the task during a rebalance should be rethrown to stop the worker
+        if (rebalanceException != null) {
```

Comment:
Hmm.. The exception can only be non-null after the call to consumer.poll(), so I'm not sure that would work. It does kind of suck that we have to wait for the timeout though, especially since we're just throwing away the fetched records in that case. That would be one argument to have the consumer propagate exceptions in the rebalance callback instead of swallowing them.